snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -526,12 +524,15 @@ FileBasedSnapshot persistNewSnapshot(
 
     try (final var ignored = snapshotMetrics.startPersistTimer()) {
       // it's important to persist the checksum file only after the move is finished, since we use
-      // it
-      // as a marker file to guarantee the move was complete and not partial
+      // it as a marker file to guarantee the move was complete and not partial
       final var destination = buildSnapshotDirectory(snapshotId);
       final var checksumPath = buildSnapshotsChecksumPath(snapshotId);
+      final var tmpChecksumPath =
+          checksumPath.resolveSibling(checksumPath.getFileName().toString() + TMP_CHECKSUM_SUFFIX);
       try {
-        SnapshotChecksum.persist(checksumPath, immutableChecksumsSFV);
+        SnapshotChecksum.persist(tmpChecksumPath, immutableChecksumsSFV);
+        Files.move(tmpChecksumPath, checksumPath, StandardCopyOption.ATOMIC_MOVE);
```

Comment:
:x: I think we need to handle `AtomicMoveNotSupportedException` here. Just falling back to a non-atomic move is probably the only/best option?